src/utils/ContainerUtils.ts

```diff
@@ -0,0 +1,97 @@
+import { ContainerProps, ListView } from "../components/CheckBoxFilterContainer";
+
+export const parseStyle = (style = ""): {[key: string]: string} => {
+    try {
+        return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+            const pair = line.split(":");
+            if (pair.length === 2) {
+                const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                styleObject[name] = pair[1].trim();
+            }
+            return styleObject;
+        }, {});
+    } catch (error) {
+        // tslint:disable-next-line no-console
+        window.console.log("Failed to parse style", style, error);
+    }
+
+    return {};
+};
+
+export class Utils {
+    static validateProps(props: ContainerProps): string {
+        const widgetName = props.friendlyId;
+        const errorMessage = [];
+
+        if (props.filterBy === "XPath" && !props.constraint) {
+            errorMessage.push(`Filter by 'XPath' requires an 'XPath contraint'`);
+        }
+        if (props.filterBy === "attribute" && !props.attribute) {
+            errorMessage.push(`Filter by 'Attribute' requires an 'Attribute'`);
+        }
+        if (props.filterBy === "attribute" && !props.attributeValue) {
+            errorMessage.push(`Filter by 'Attribute' requires an 'Attribute value`);
+        }
+        if (props.unCheckedFilterBy === "XPath" && !props.unCheckedConstraint) {
+            errorMessage.push(`Unchecked filter by 'XPath' requires an 'XPath contraint'`);
+        }
+        if (props.unCheckedFilterBy === "attribute" && !props.unCheckedAttribute) {
+            errorMessage.push(`Unchecked filter by 'Attribute' requires an 'Attribute'`);
+        }
+        if (props.unCheckedFilterBy === "attribute" && !props.unCheckedAttributeValue) {
+            errorMessage.push(`Unchecked filter by 'Attribute' requires an 'Attribute value`);
+        }
+        if (errorMessage.length) {
+            return `${widgetName} : ${errorMessage.join(", ")}`;
+        }
+
+        return "";
+    }
+
+    static validateCompatibility(props: ContainerProps & { targetListView: ListView; }): string {
+        const { targetListView } = props;
+        const type = targetListView && targetListView.datasource.type;
+        const widgetName = props.friendlyId;
+
+        if (!targetListView) {
+            return `${widgetName}: unable to find a listview with to attach to`;
+        }
+
+        if (type && type !== "database" && type !== "xpath") {
+            return `${widgetName}, widget is only compatible with list view data source type 'Database' and 'XPath'`;
+        }
+        if (!(targetListView && targetListView._datasource && targetListView._entity && targetListView.update)) {
```

Comment:
Please double check your code.... you are missing some checks for private API

Reply:
Was handled in `mendix-data-source-helper`